sdk/src/cose_sign.rs

```diff
@@ -148,6 +162,73 @@ pub fn cose_sign(signer: &dyn Signer, data: &[u8], box_size: usize) -> Result<Ve
     Ok(c2pa_sig_data)
 }
 
+/// Returns signed Cose_Sign1 bytes for "data".  The Cose_Sign1 will be signed with the algorithm from `Signer`.
+#[cfg(feature = "async_signer")]
+pub async fn cose_sign_async(
```

Comment:
Can the cose_sign functionality be moved to a private method to avoid code duplication